### ISSUE TITLE
NativeSelect setEmptySelectionCaption does not work for non-String NativeSelects

### ISSUE THREAD

**Issue by eriklumme** [vaadin/framework#10937]
*Steps to reproduce*
- Create a new `NativeSelect<Integer>`
- Set the empty selection caption to "Select"
- Add some numbers to the NativeSelect

*Expected outcome*
![image](https://user-images.githubusercontent.com/29859656/40344833-4e7d64f2-5d9f-11e8-9ecf-c9668b18eb24.png)
![image](https://user-images.githubusercontent.com/29859656/40344853-5e91fef2-5d9f-11e8-9019-62ea7c6143ef.png)

*Actual outcome*
![image](https://user-images.githubusercontent.com/29859656/40344868-6c81495a-5d9f-11e8-9e2a-7faa8d46e7fe.png)
![image](https://user-images.githubusercontent.com/29859656/40344877-76b054ac-5d9f-11e8-9deb-d2c53ded763b.png)

So basically, the empty selection caption occurs in the list, but it's not selected by default, and can not be selected by server-side code. Using `clear()`, `setValue(null)` or `setSelectedItem(null)` sets the NativeSelect back to the empty string caption.

Most likely, this is due to the empty selection caption being shown when an empty string "" is selected, not when null is selected.

Same issue as here (permalink to second to last message):  https://vaadin.com/forum/thread/15654110/16909551

---

Framework version: 8.4.2
Browser version: Chrome 59.0.3071.86

**Comment by eriklumme** [comment]
Similar issue to what was fixed for ComboBox here https://github.com/vaadin/framework/issues/9079

**Comment by anasmi** [comment]
That actually looks correct. You will need to _setValue_ to empty. 

`select.setSelectedItem("");`

should work and you will see the caption.

**Comment by mstahv** [comment]
The setEmptySelectionCaption seems completely useless to me because of this bug.
